cmake: link ceph-fuse against librt

wy7980 opened the pull request:
ceph-fuse: fix error of building ceph-fuse
fix linking error of without -lrt for ceph-fuse.

error message:
```
[100%] Building CXX object src/CMakeFiles/ceph-fuse.dir/ceph_fuse.cc.o
[100%] Building CXX object src/CMakeFiles/ceph-fuse.dir/client/fuse_ll.cc.o
[100%] Linking CXX executable ../bin/ceph-fuse
/opt/rh/devtoolset-7/root/usr/libexec/gcc/x86_64-redhat-linux/7/ld: CMakeFiles/ceph-fuse.dir/ceph_fuse.cc.o: undefined reference to symbol 'clock_gettime@@GLIBC_2.2.5'
//lib64/librt.so.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[3]: *** [bin/ceph-fuse] Error 1
make[2]: *** [src/CMakeFiles/ceph-fuse.dir/all] Error 2
make[1]: *** [src/CMakeFiles/ceph-fuse.dir/rule] Error 2
make: *** [ceph-fuse] Error 2
```

Signed-off-by: Yong Wang  <wy7980@sina.com>


<!--
Thank you for opening a pull request!  Here are some tips on creating
a well formatted contribution.

Please give your pull request a title like "[component]: [short description]"

This is the format for commit messages:

"""
[component]: [short description]

[A longer multiline description]

Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
Signed-off-by: [Your Name] <[your email]>
"""

The Signed-off-by line is important, and it is your certification that
your contributions satisfy the Developers Certificate or Origin.  For
more detail, see SubmittingPatches.rst.

The component is the short name of a major daemon or subsystem,
something like "mon", "osd", "mds", "rbd, "rgw", etc. For ceph-mgr modules,
give the component as "mgr/<module name>" rather than a path into pybind.

For more examples, simply use "git log" and look at some historical commits.

This was just a quick overview.  More information for contributors is available here:
https://raw.githubusercontent.com/ceph/ceph/master/SubmittingPatches.rst

-->
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

tchaikov commented on src/CMakeLists.txt:
```diff
@@ -689,7 +689,7 @@ if(WITH_FUSE)
   add_executable(ceph-fuse ${ceph_fuse_srcs})
   target_link_libraries(ceph-fuse ${FUSE_LIBRARIES}
-    ${GSSAPI_LIBRARIES} client ceph-common global-static)
+    ${EXTRALIBS} ${GSSAPI_LIBRARIES} client ceph-common global-static)
```
shall we move `${EXTRALIBS}` after `ceph-common` or even after `global-static`? i am afraid that some compilation units in them might reference `clock_gettime()` and other symbols defined by `${EXTRALIBS}`.

wy7980 replied:
That's ok, i add the item just refer to the use of it in src/CMakeLists.txt and build it successfully.

wy7980 commented:
> i'd suggest revise the title of commit message to something like
> 
> > cmake: link ceph-fuse against librt

Yes, the new title can be more clear.

tchaikov commented:
@wy7980 could you squash these two commits into a single one?

wy7980 commented:
> @wy7980 could you squash these two commits into a single one?

No problem.

Please look at the new one https://github.com/ceph/ceph/pull/31549

tchaikov commented:
@wy7980 instead of creating a new PR for addressing the reviewers' comments, i'd recommend reusing the existing one. you can always use `git push -f` for updating your remote branch.

tchaikov commented:
@wy7980 also, would be great if you could use your full name in your "Signed-off-by" line. i don't think it's "wy7980", otherwise it's just very unusual..

wy7980 commented:
> @wy7980 instead of creating a new PR for addressing the reviewers' comments, i'd recommend reusing the existing one. you can always use `git push -f` for updating your remote branch.

got it.
